compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -7548,15 +7506,17 @@ OMR::ARM64::TreeEvaluator::arraycopyEvaluator(TR::Node *node, TR::CodeGenerator
       stopUsingCopyReg5 = true;
       }
 
-   // x0-x4 are destroyed in the helper
-   TR::RegisterDependencyConditions *deps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(5, 5, cg->trMemory());
+   // x0-x3 and v0-v1 are destroyed in the helper
+   TR::RegisterDependencyConditions *deps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(6, 6, cg->trMemory());
    TR::addDependency(deps, lengthReg, TR::RealRegister::x0, TR_GPR, cg);
    TR::addDependency(deps, srcAddrReg, TR::RealRegister::x1, TR_GPR, cg);
    TR::addDependency(deps, dstAddrReg, TR::RealRegister::x2, TR_GPR, cg);
    TR::addDependency(deps, NULL, TR::RealRegister::x3, TR_GPR, cg);
-   TR::addDependency(deps, NULL, TR::RealRegister::x4, TR_GPR, cg);
+   TR::addDependency(deps, NULL, TR::RealRegister::v0, TR_FPR, cg);
+   TR::addDependency(deps, NULL, TR::RealRegister::v1, TR_FPR, cg);
```

Comment:
I wonder if it is better to choose less popular registers to trash in the helper (e.g., v30/v31).  The reason being that the local register assigner will often choose v0/v1 (won't it?) which may be live across this point, and this increases the likelihood of a conflict that would get resolved by evicting the contents of those registers across this call.

Reply:
Agreed.
I updated the code to use v30 and v31 instead of v0 and v1.  They appear as q30 and q31 in ARM64arrayCopy.spp.

Reply:
I ran OpenJ9 sanity.functional tests with this update in https://github.com/eclipse-openj9/openj9/pull/19470, and got no failures.